pytradfri/api/aiocoap_api.py

```diff
@@ -175,14 +188,46 @@ def psk(self, value):
 
         def success_callback(res):
             api_command.result = _process_output(res)
+            APIFactory.update_last_changed()
 
         def error_callback(ex):
             err_callback(ex)
 
         ob = pr.observation
         ob.register_callback(success_callback)
         ob.register_errback(error_callback)
-        self._observations_err_callbacks.append(ob.error)
+        self._observations.append(ob)
+
+    async def _check_observations(self):
+        if self._is_checking:
+            _LOGGER.debug("Already checking for observations...")
+            return
+
+        self._is_checking = True
+        current_time = time.time()
+        await asyncio.sleep(OBSERVATION_SLEEP_TIME, loop=self._loop)
+
+        if (current_time - APIFactory.get_last_changed()) > \
+                (OBSERVATION_TIMEOUT + OBSERVATION_SLEEP_TIME):
+            _LOGGER.warning('Resetting Tradfri observations...')
+
+            if self._is_resetting:
+                return
+
+            self._is_resetting = True
+
+            while self._observations:
+                ob = self._observations.pop()
```

Comment:
Do we need to let aiocoap know that we're ending the observation? If we discard an active observation accidentally does that mean we'll potentially leak a connection?

Reply:
The connections should end as defined in `aiocoap.protocol._BaseClientObservation`. I've reworked this a bit in 17638af0ceb00e40273710bec419086207a3318b so it should also cancel out.